messaging/src/main/java/org/axonframework/commandhandling/gateway/DefaultNonTransientPredicate.java

```diff
@@ -0,0 +1,16 @@
+package org.axonframework.commandhandling.gateway;
+
+import org.axonframework.common.AxonNonTransientException;
+
+/**
+ * A default predicate for checking non-transiency of a failure comparing it against concrete classes.
+ * <p/>
+ * This default implementation uses only {@link AxonNonTransientException} class for comparisons.
+ * <p/>
+ * This predicate is intended to be used for configuring {@link RetryScheduler} implementations.
+ */
+public class DefaultNonTransientPredicate extends NonTransientClassesPredicate {
```

Comment:
Wouldn't renaming this class to `AxonNonTransientExceptionPredicate` make more sense? 
Sure, it's the default, but that might change in the future. If so, the name "default" would be off, of course.

Reply:
It makes sense. However, I see now that class names are a bit off. How about `NonTransientExceptionClassesPredicate` instead of `NonTransientClassesPredicate` and `AxonNonTransientExceptionClassesPredicate` instead of `AxonNonTransientExceptionPredicate`. 
It's a bit longer, but more to the point. WDYT?

Reply:
I added those longer names to the latest commits. Please see if this is ok.

Reply:
Sounds good atm. I'll re-review the adjustments tomorrow in more detail :+1: